### PR TITLE
Updated package.json with typescript v5.0 and some little fixes

### LINKED ISSUE #1189: Installation fails often with npm/pnpm because typescript-eslint🐛 and some others errors [Bug report]:
### Descrcibe the bug?

Some people as myself noted that sometimes, we can't install the application because an error happens with typescript-eslint because the module is missing but it's present in the package.son file.

No problem when the installation is done with Yarn nevertheless.

### Steps to reproduce the bug?

Clone the repository.
Install with pnpm
Start the application

We got this message
image

In the same way, we got problems with the process.env variables that are not found 

image

or some missing icons from react-icons.
image

### Expected behavior

Installation needs to be done with every package manager without errors.

### Desktop (Please provide your system information)

Device : Laptop Hp
Windows 11
browser : Vivaldi / Chrome (@latest)  

### Mobile (Please provide your device information)

_No response_

### Screenshot / Screenshare

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Angra974**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

Problem 1: 
Installation often fails when installed with pnpm.

We got this error.
image

Reactplay use Eslint v8 and this version need at least typescript v5.
Solution :
As the requirement with another parameters need typescript < 5.1.0, I put the version to v5.0 and no warnig are shown for this issue.

Problem 2: 
process if not always found. ( i only use in the application in develpment so i don't talk for production here)
image

The problem is because @types/node is missing.
Solution: add the dev dependencies

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Problem 3 : 
A type is missing in one play, and it gives a warning,
image

Solution: add the @types/p5 

Problem 4: 
lodash.memoize is needed for a play. but not installed.
Solution : Ask the owner to modifify the play or add the dependencies.
image
I've added the dependencies as the play is allready present.

Problem 5:
Some icons are not found in react-icons and generate some errors.
image
Solution: An update to react-icons v4.8.0 seems to resolve the issue or generate fewer problems as no issue are shown on my part when resolving the issue.

Other:
Added @babel/plugin-proposal-private-property-in-object as a dev dependency to suppress a warning.
Some discussion about the problem can be found here => https://stackoverflow.com/questions/76435306/babel-preset-react-app-is-importing-the-babel-plugin-proposal-private-propert

From Cypress build : 
image

Fixes #1189 

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Downloaded the last version.
First installation done with pnpm.
Do all the required modification/updates in packageJson by installing the package related to the warning.
Test the application after it runs well.
Copy all the files to a new folder
Second installation with yarn.
Check the result if all works as expected.

# Checklist:

- [X] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by atapas** [comment]
@supminn @Sachin-chaurasiya @ap221882 please take a look

**Review by Sachin-chaurasiya — APPROVED** [review]
Looks Good 👍

File: package.json [hunk 1]
```diff
@@ -37,9 +37,10 @@
     "jspdf": "^2.5.1",
     "leaflet": "^1.9.3",
     "lodash": "^4.17.21",
+    "lodash.memoize": "^4.1.2",
     "mathjs": "^11.5.1",
     "node-sass": "^8.0.0",
-    "p5": "^1.5.0",
+    "p5": "^1.3.1",
```
Comment (supminn) [inline package.json:43]: Why is the version for p5 downgraded?

Comment (Angra974) [reply package.json:43]: Check the dependencies of react-p5. While using 1.5.0, they are a problem of type. You can resolve it by using the same dependencies needs by react-p5.

https://github.com/Gherciu/react-p5/blob/main/package.json

**Review by supminn — APPROVED** [review]
These changes LTGM 👍🏻

**Review by ap221882 — COMMENTED** [review]
It works with `pnpm install`. But there are complains when we install using `npm install --legacy-peer-deps`. If that can be resolved will be of many help.

**Comment by Angra974** [comment]
> It works with `pnpm install`. But there are complains when we install using `npm install --legacy-peer-deps`. If that can be resolved will be of many help.

Do you had a screenshot to add to this so i can try to look into it.

**Comment by Angra974** [comment]
> It works with `pnpm install`. But there are complains when we install using `npm install --legacy-peer-deps`. If that can be resolved will be of many help.

Just got some depreciated warning info more than before but nothing serious.

Actual Reactplay :
image

This branch :
image

each has the same number of vulnerabilities ( same for moderate, high, and critical ).
Some can be resolved by updating other dependencies too but not in this submission as workbox is now at v7 and we are using 6.5.4 and you have a depreciated warning for workbox 6.6.0.

**Comment by atapas** [comment]
@ap221882 it's not approved by you. Is there anything pending?

**Comment by atapas** [comment]
@Angra974 there are merge conflicts could you please take care.

**Comment by ap221882** [comment]
@atapas I am approving this but I was getting errors when not using pnpm. Also if conflicts are resolved we are good for now.

**Review by ap221882 — APPROVED** [review]

**Pull request closed without merging**